server/service/src/sync/translations/mod.rs

```diff
@@ -199,93 +199,107 @@ pub(crate) trait SyncTranslation {
         Ok(None)
     }
 
-    fn try_translate_push(
+    /// Implementation should return three types of results
+    /// * Error - Something completely unexpected that is not recoverable
+    /// * None - Translator did not match record type
+    /// * Some - Translator did match and either translated record/records or
+    ///          empty array if record is deliberatly ignored
+    fn try_translate_push_upsert(
         &self,
         _: &StorageConnection,
         _: &ChangelogRow,
-    ) -> Result<Option<Vec<PushUpsertRecord>>, anyhow::Error> {
+    ) -> Result<Option<Vec<RemoteSyncRecordV5>>, anyhow::Error> {
         Ok(None)
     }
-}
 
-#[derive(Debug)]
-pub(crate) struct PushUpsertRecord {
-    pub(crate) sync_id: i64,
-    /// The translated table name
-    pub(crate) table_name: &'static str,
-    pub(crate) record_id: String,
-    pub(crate) data: serde_json::Value,
+    fn try_translate_push_delete(
+        &self,
+        _: &StorageConnection,
+        _: &ChangelogRow,
+    ) -> Result<Option<Vec<RemoteSyncRecordV5>>, anyhow::Error> {
+        Ok(None)
+    }
 }
 
-pub(crate) struct PushDeleteRecord {
-    pub(crate) sync_id: i64,
-    /// The translated table name
-    pub(crate) table_name: &'static str,
-    pub(crate) record_id: String,
+impl RemoteSyncRecordV5 {
+    pub(crate) fn new_upsert(
+        changelog: &ChangelogRow,
+        table_name: &'static str,
+        data: serde_json::Value,
+    ) -> Self {
+        Self {
+            sync_id: changelog.cursor.to_string(),
+            record: CommonSyncRecordV5 {
+                table_name: table_name.to_string(),
+                record_id: changelog.record_id.clone(),
+                action: SyncActionV5::Update,
+                data,
+            },
+        }
+    }
+
+    pub(crate) fn new_delete(changelog: &ChangelogRow, table_name: &'static str) -> Self {
+        Self {
+            sync_id: changelog.cursor.to_string(),
+            record: CommonSyncRecordV5 {
+                table_name: table_name.to_string(),
+                record_id: changelog.record_id.clone(),
+                action: SyncActionV5::Delete,
+                data: Default::default(),
+            },
+        }
+    }
 }
 
-pub(crate) enum PushRecord {
-    Upsert(PushUpsertRecord),
-    Delete(PushDeleteRecord),
+#[derive(Error, Debug)]
+#[error("Problem translation push record: {changelog:?}")]
+pub(crate) struct PushTranslationError {
+    changelog: ChangelogRow,
+    source: anyhow::Error,
 }
 
-pub(crate) fn table_name_to_central(table: &ChangelogTableName) -> &'static str {
```

Comment:
This translation should happen in translators (one reason for refactor, I didnt' like translation logic being spread out, it's good to keep it to translators, imo)